official/benchmark/models/resnet_imagenet_main.py

```diff
@@ -186,6 +241,17 @@ def run(flags_obj):
           weights=None,
           classes=imagenet_preprocessing.NUM_CLASSES,
           layers=tf.keras.layers)
+    elif flags_obj.model == 'mobilenet_pretrained':
+      model = tf.keras.applications.mobilenet.MobileNet(
```

Comment:
> I'd dedup the code from flags_obj.model == 'mobilenet' to reflect that.

Which one is better for the 'dedup' here? 

```
    elif flags_obj.model == 'mobilenet' or 'mobilenet_pretrained':
      # TODO(kimjaehong): Remove layers attribute when minimum TF version
      # support 2.0 layers by default.
      if flags_obj.model == 'mobilenet_pretrained':
        classes_labels = 1000
        dropout_rate = 1e-7
        initial_weights = 'imagenet'
      else:
        classes_labels = imagenet_preprocessing.NUM_CLASSES
        dropout_rate = 1e-3
        initial_weights = None
      model = tf.keras.applications.mobilenet.MobileNet(
          dropout=dropout_rate,
          weights=initial_weights,
          classes=classes_labels,
          layers=tf.keras.layers)
```
or 

```
    elif flags_obj.model == 'mobilenet':
      # TODO(kimjaehong): Remove layers attribute when minimum TF version
      # support 2.0 layers by default.
      model = tf.keras.applications.mobilenet.MobileNet(
          dropout=1e-3,
          weights=None,
          classes=imagenet_preprocessing.NUM_CLASSES,
          layers=tf.keras.layers)
    elif flags_obj.model == 'mobilenet_pretrained':
      model = tf.keras.applications.mobilenet.MobileNet(
          dropout=1e-7,
          weights='imagenet',
          classes=1000,
          layers=tf.keras.layers)    
```
Or, do you have anything else in your mind?

Reply:
First one is preferable. I was wondering though why the # of labels and dropout_rate need to differ. For the case of the dropout_rate, does it actually affect the resulting accuracy significantly to not use the default?

Reply:
yea, the dropout_rate and # of labels do affect the resulting accuracy as I recall. However, I cannot find the test results of how much of a difference they are making. I can rerun the tests if necessary?

Reply:
Similarly above, I see the branch in the optimizer to set a different learning rate. 

Does the experiment not work if we use the existing flags_obj.initial_learning_rate_per_sample * flags_obj.batch_size instead?

Reply:
Ok. It could be that the pretrained weights themselves work with 1000 classes. I think it's fine to defer that to the future.

For dropout and the optimizer, I'd see if we can remove the branch though and run tests to see if they matter.

Reply:
If they do, maybe it'll be preferable to change the dropout for pruning to match clustering. You can leave a TODO for tfmot for that.

Reply:
> Does the experiment not work if we use the existing flags_obj.initial_learning_rate_per_sample * flags_obj.batch_size instead?

It works. I tested that and we can remove the optimizer branch.

For dropout, however, my coworker got different results from mine. I will double-check it next week (Haven't got time to look at this work for the past week. Sorry!).

Reply:
So we have run the script `resnet_imagenet_main.py` with the dropout equals to the default value several times and the accuracy was good, and the maximum variation is **0.26%**. However, when I use `tf.distribute` to test the same script, it seems that the average accuracy is consistently lower and the maximum variation is **0.73%**.

| tf.distribute  | run 1 | run 2 | run 3 | run 4 | run 5 | maximum variation
|----------------|-------|-------|-------|-------|-------|-------|
| on with 1 GPU  | 0.6907 | 0.6906 | 0.6836 | 0.6875 | 0.6909 | 0.0073|
| off with 1 GPU | 0.6939 | 0.6944 | 0.6951 | 0.6965 | 0.6964 | 0.0026 |

I think the test results are acceptable in both cases and they proved that the dropout can be set to the default value. However, the effect of `tf.distribute` on the test results can be investigated in other tasks, perhaps? @alanchiao

Reply:
Fine with investigating in a separate PR.